astropy/coordinates/solar_system.py

```diff
@@ -56,116 +68,91 @@ def _download_spk_file(url=('http://naif.jpl.nasa.gov/pub/naif/'
 
     .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
     """
+    de_number = int(ephemeris[2:])
+    if not ephemeris.startswith('de') or not (0 < de_number < 1000):
+        raise ValueError("JPL ephemeris names be 'deNNN'.")
+    url = ('http://naif.jpl.nasa.gov/pub/naif/'
+           'generic_kernels/spk/planets/de{0:03d}.bsp'.format(de_number))
+
     return download_file(url, cache=True, show_progress=show_progress)
 
 
-def _get_kernel(*args, **kwargs):
+def _get_kernel(ephemeris='de430'):
     """
     Try importing jplephem, download/retrieve from cache the Satellite Planet
-    Kernel.
+    Kernel corresponding to the given ephemeris.
     """
     global KERNEL
 
-    try:
-        from jplephem.spk import SPK
-    except ImportError:
-        raise ImportError("Solar system ephemeris calculations require the "
-                          "jplephem package "
-                          "(https://pypi.python.org/pypi/jplephem)")
+    ephemeris = ephemeris.lower()
+
+    if KERNEL is None or KERNEL.name != ephemeris:
+        try:
+            from jplephem.spk import SPK
+        except ImportError:
+            raise ImportError("Solar system JPL ephemeris calculations require "
+                              "the jplephem package "
+                              "(https://pypi.python.org/pypi/jplephem)")
 
-    if KERNEL is None:
-        KERNEL = SPK.open(_download_spk_file())
+        KERNEL = SPK.open(_download_spk_file(ephemeris))
+        KERNEL.name = ephemeris
     return KERNEL
 
 
-def get_body_barycentric(time, body):
+def get_body_barycentric(body, time, ephemeris=None):
     """
     Calculate the barycentric position of the solar system body ``body``
     in cartesian coordinates.
 
-    Uses ``jplephem`` with the DE430 kernel.
-
     Parameters
     ----------
-    time : `~astropy.time.Time`
-        Time of observation
-
     body : str
-        The solar system body to calculate.
-
-        The allowed values for ``body`` can be found in
+        The solar system body to calculate. The allowed values for ``body``
+        depend on the ephemeris, but standard ones can be found in
```

Comment:
Yes, I realised that my nice suggestion was no longer so nice... If we do have a "science state" than this would be easy to update. I think we can also state in the docstring some objects that are always available.